### PR TITLE
ISIS: LSP: Add extened IS + IP reachability TLVs

### PR DESCRIPTION
**Pull request by taktv6**

### CODE REVIEW

File: protocols/isis/server/lsp.go [hunk 1]
```diff
@@ -39,8 +39,8 @@ func (s *Server) generateLocalLSP() *packet.LSPDU {
 			packet.NewAreaAddressesTLV(s.areaIDs()),
 			s.getProtocolsSupportedTLV(),
 			packet.NewIPInterfaceAddressesTLV(s.netIfaManager.getAddressesIPv4()),
-			// TODO: Add ExtendedISReachabilityTLV
-			// TODO: Add ExtendedIPReachabilityTLV
+			s.ExtendedIPReachabilityTLV(),
```
Comment (BarbarossaTM) [inline protocols/isis/server/lsp.go:42]: Is there a reason why one is exported and the other isn't?

File: protocols/isis/server/neighbor.go [hunk 2]
```diff
@@ -230,3 +230,26 @@ func getP2PAdjTLV(tlvs []packet.TLV) *packet.P2PAdjacencyStateTLV {
 func (n *neighbor) p2pAdjTLVContainsSelf(t *packet.P2PAdjacencyStateTLV) bool {
 	return t.NeighborSystemID == n.nm.netIfa.srv.nets[0].SystemID && t.NeighborExtendedLocalCircuitID == uint32(n.nm.netIfa.devStatus.GetIndex())
 }
+
+func (n *neighbor) extendedISReachabilityNeighbor() *packet.ExtendedISReachabilityNeighbor {
```
Comment (BarbarossaTM) [inline protocols/isis/server/neighbor.go:234]: Do you want to tackle TODO + FIXME later?

Comment (taktv6) [reply protocols/isis/server/neighbor.go:234]: Leaving the TODO in. Fixed the FIXME.

File: protocols/isis/server/lsp.go (hunk 1 above)
Comment (taktv6) [reply protocols/isis/server/lsp.go]: No. Just me being inattentive. Fixed.

**Review by BarbarossaTM — APPROVED** [review]
🦆

**Pull request merged**